### PR TITLE
[Sprint 7] Implement API TimeSheet

### PR DESCRIPTION
**Pull request by nguyenminhduong**
Redime https://dev.framgia.com/redmine/issues/126344
Implement API TimeSheet, fix timesheet ui

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java [hunk 1]
```diff
@@ -130,11 +130,30 @@ public boolean isColorMorning() {
...
+    }
+
     public boolean isColorAllDay() {
         return StringUtils.isNotBlank(mColorAfternoon) && StringUtils.isNotBlank(mColorMorning);
     }
+
+    private String convertColor(String color) {
+        String result = "#";
+        int i = 1;
+        while (i < color.length()) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java]: you can try it

if(TextUtils.isEmpty(colorString)) return null;
if(colorString.length() == 4) { 
colorString = "#" 
"+ StringUtils.repeat(colorString.substring(1, 1), 1) 
"+ StringUtils.repeat(colorString.substring(2, 2), 1) 
"+ StringUtils.repeat(colorString.substring(3, 3), 1) 
},

File: app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java [hunk 2]
```diff
@@ -92,19 +92,19 @@ public void setTextAfternoon(String textAfternoon) {
...
-        return mColorMorning;
-    }
-
-    public void setColorMorning(String colorMorning) {
-        mColorMorning = colorMorning;
+        // todo edit later
+        if (mColorMorning == null) {
+            return null;
+        }
+        return mColorMorning.length() == 4 ? convertColor(mColorMorning) : mColorMorning;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java]: Pls declare ``4`` as a constant variable

File: app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java [hunk 3]
```diff
@@ -92,19 +92,19 @@ public void setTextAfternoon(String textAfternoon) {
...
-        return mColorAfternoon;
-    }
-
-    public void setColorAfternoon(String colorAfternoon) {
-        mColorAfternoon = colorAfternoon;
+        // todo edit later
+        if (mColorAfternoon == null) {
+            return null;
+        }
+        return mColorAfternoon.length() == 4 ? convertColor(mColorAfternoon) : mColorAfternoon;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/model/TimeSheetDate.java]: like above

File: app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java [hunk 4]
```diff
@@ -47,12 +49,13 @@ public TimeSheetViewModel(Context context, TimeSheetContract.Presenter presenter
...
         mTimeSheetDates = new ArrayList<>();
         mTimeSheetDate = new TimeSheetDate();
+        mUserTimeSheet = new UserTimeSheet();
         mDialogManager = dialogManager;
         mNavigator = navigator;
         mDialogManager.showIndeterminateProgressDialog();
         setMonth();
         setYear();
-        mPresenter.getTimeSheet(mMonth, mYear);
+        mPresenter.getTimeSheet(mMonth + 1, mYear);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/timesheet/TimeSheetViewModel.java]: pls define 1 as a constant

**Pull request merged**